src/main/java/org/sonarsource/sonarlint/ls/AnalysisManager.java

```diff
@@ -440,20 +429,31 @@ private static Range position(Issue issue) {
         issue.getEndLineOffset()));
   }
 
-  private static Range position(IssueLocation location) {
-    return new Range(
-      new Position(
-        location.getStartLine() - 1,
-        location.getStartLineOffset()),
-      new Position(
-        location.getEndLine() - 1,
-        location.getEndLineOffset()));
+  static String message(Issue issue) {
+    if (issue.flows().isEmpty()) {
+      return issue.getMessage();
+    } else if (issue.flows().size() == 1) {
```

Comment:
I think there is a mistake here. "Secondary" locations are stored as separate flows, each having a single location.
This should be `if (issue.flows().stream().allMatch(f -> f.size() ==1))`